### PR TITLE
Use kubernetes client waitUntilReady instead of the for loop

### PR DESCRIPTION
**Pull request by carlossg**

### CODE REVIEW

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java [hunk 1]
```diff
@@ -692,77 +694,46 @@ public Node call() throws Exception {
...
 
-                LOGGER.log(Level.FINE, "Creating Pod: {0} in namespace {1}", new Object[] { podId, namespace });
+                String[] logArgs = new String[] { namespace, podId };
+                LOGGER.log(Level.FINE, "Creating Pod: {0}/{1}", logArgs);
                 pod = client.pods().inNamespace(namespace).create(pod);
-                LOGGER.log(Level.INFO, "Created Pod: {0} in namespace {1}", new Object[] { podId, namespace });
 
-                // We need the pod to be running and connected before returning
-                // otherwise this method keeps being called multiple times
-                ImmutableList<String> validStates = ImmutableList.of("Running");
```
Comment (marvinthepa) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java]: This was the only instance of `ImmutableList` in use, you can remove the import.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/ProvisioningCallback.java [hunk 2]
```diff
@@ -145,73 +147,44 @@ public Node call() throws Exception {
...
-                                    new Object[] { podId, info.getState().getWaiting(), info.getName() });
-                            // break;
-                        }
-                        if (info.getState().getTerminated() != null) {
-                            terminatedContainers.add(info);
-                        } else if (!info.getReady()) {
-                            allContainersAreReady = false;
+            LOGGER.log(Level.INFO, "Waiting for pod to be ready: {0}/{1}", logArgs);
+            try {
+                client.pods().inNamespace(namespace).withName(podId).waitUntilReady(10, TimeUnit.MINUTES);
```
Comment (marvinthepa) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/ProvisioningCallback.java]: Wouldn't this be a good opportunity to make this wait time configurable, at least as a system property?

10 minutes seems a bit log for starting up a slave pod (especially if you want to manually test the error handling code 😁 ).

Comment (carlossg) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/ProvisioningCallback.java]: I think it was 10 minutes already

**Comment by marvinthepa** [comment]
N.B. another problem with the "typo in image name" is that kubernetes will actually try a few times with the error above, and will then change into "ImagePullBackoff" - without retaining the error from `ErrImagePull` before:

```
Container is waiting pod-test-0001-5l7w2-zc1cr [devx]: ContainerStateWaiting(message=Back-off pulling image "registry:5005/bla/blubb", reason=ImagePullBackOff, additionalProperties={})
```

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/ProvisioningCallback.java (hunk 2 above)
Comment (carlossg) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/ProvisioningCallback.java]: needs to be in MILLISECONDS due to a bug in the k8s client

**Pull request closed without merging**